libs/execution_context/ebpf_maps.c

```diff
@@ -1067,20 +1165,26 @@ _insert_into_hot_list(_Inout_ ebpf_core_lru_map_t* map, _Inout_ ebpf_lru_entry_t
  * @param[in] key Key to initialize the entry with.
  */
 static void
-_initialize_lru_entry(_Inout_ ebpf_core_lru_map_t* map, _Inout_ ebpf_lru_entry_t* entry, _In_ const uint8_t* key)
+_initialize_lru_entry(
+    _Inout_ ebpf_core_lru_map_t* map, _Inout_ ebpf_lru_entry_t* entry, size_t partition, _In_ const uint8_t* key)
 {
-    ebpf_lock_state_t state = ebpf_lock_lock(&map->lock);
-    ebpf_assert(_get_key_state(map, entry) == EBPF_LRU_KEY_UNINITIALIZED);
+    memcpy(EBPF_LRU_ENTRY_KEY_PTR(map, entry), key, map->core_map.ebpf_map_definition.key_size);
```

Comment:
For my understanding, can you help clarify the purpose of doing memcpy here?
And also, I think lock is needed during initialization, to protect this critical section.

Reply:
Each value stored in the hash-table has a supplemental value associated with it. These supplemental values are used to build the key-history lists. Given a supplemental value, there is no easy way to determine what the original key is, so we copy the key value into the supplemental value field.